windows-installer-agent.ps1

```diff
@@ -0,0 +1,148 @@
+    } catch {
+        Write-Warning "Error removing existing service: $_"
+        # Continue anyway since we'll try to create a new one
+    }
+}
+
+# Remove old agent if it exists
+if (Test-Path $agentBinPath) {
+    if (-not $Silent) { Write-Host "Removing old agent..." }
+    try {
+        Remove-Item -Path $agentBinPath -Force
+    } catch {
+        Write-Error "Failed to remove old agent: $_"
+        exit 1
+    }
+}
+
+# Create directories if they don't exist
+if (Test-Path $cachePath) {
+    if (-not $Silent) { Write-Host "Cleaning up existing cache directory..." }
+    Remove-Item -Path $cachePath -Recurse -Force
+}
+
+if (Test-Path $binPath) {
+    if (-not $Silent) { Write-Host "Cleaning up existing bin directory..." }
+    Remove-Item -Path $binPath -Recurse -Force
+}
+
+if (-not $Silent) { Write-Host "Creating cache directory..." }
+New-Item -ItemType Directory -Path $cachePath -Force | Out-Null
+
+if (-not $Silent) { Write-Host "Creating bin directory..." }
+New-Item -ItemType Directory -Path $binPath -Force | Out-Null
+
+# Download the agent
+if (-not $Silent) { Write-Host "Downloading Viam Agent..." }
+try {
+    Invoke-WebRequest -Uri "https://storage.googleapis.com/packages.viam.com/temp/$agentCURLFileName" -OutFile $agentCachePath
+    if (-not (Test-Path $agentCachePath)) {
+        throw "Failed to download agent executable."
+    }
+} catch {
+    Write-Error "Failed to download Viam Agent: $_"
+    exit 1
+}
+
+if (-not $Silent) { Write-Host "Download completed successfully." }
+
+# Create symbolic link
+if (-not $Silent) { Write-Host "Creating symbolic link..." }
+try {
+    $linkArgs = @("`"$agentBinPath`"", "`"$agentCachePath`"")
+    Start-Process -FilePath "cmd.exe" -ArgumentList "/c mklink $linkArgs" -NoNewWindow -Wait
+} catch {
+    Write-Error "Failed to create symbolic link: $_"
+    exit 1
+}
+
+
+# Configure firewall
+if (-not $Silent) { Write-Host "Configuring firewall..." }
+try {
+    $firewallArgs = @(
+        "advfirewall", "firewall", "add", "rule", 
+        "name=`"$agentBinPath`"", 
+        "dir=in", 
+        "action=allow", 
+        "program=`"$agentBinPath`"", 
+        "enable=yes"
+    )
+    Start-Process -FilePath "netsh" -ArgumentList $firewallArgs -NoNewWindow -Wait
+} catch {
+    Write-Warning "Failed to configure firewall: $_"
+    # Continue despite firewall error
+}
+# Configure and start service
+if (-not $Silent) { Write-Host "Configuring service..." }
+try {
+    # Create service
+    $scArgs = @("create", "viam-agent", "binpath=", "`"$agentBinPath`"", "start=", "auto")
+    Start-Process -FilePath "sc" -ArgumentList $scArgs -NoNewWindow -Wait
+    
+    # Configure failure actions
+    $scFailArgs = @("failure", "viam-agent", "reset=", "0", "actions=", "restart/30000/restart/30000/restart/30000")
+    Start-Process -FilePath "sc" -ArgumentList $scFailArgs -NoNewWindow -Wait
+    
+    # Set failure flag
+    $scFlagArgs = @("failureflag", "viam-agent", "1")
```

Comment:
not a blocker for this PR, but in the future we'll need to think carefully through the logic here and decide:
- when agent-win should return 0
- how do we 'keep it running at all times' (unless someone asks to shut it down). not straightforward I think bc there is no simple way to request a restart

Reply:
good point, i'll think about this next weekk